hummingbot/connector/utils.py

```diff
@@ -51,3 +54,40 @@ def split_hb_trading_pair(trading_pair: str) -> Tuple[str, str]:
 def combine_to_hb_trading_pair(base: str, quote: str) -> str:
     trading_pair = f"{base}-{quote}"
     return trading_pair
+
+
+def get_new_client_order_id(
+    is_buy: bool, trading_pair: str, hbot_order_id_prefix: str = "", max_id_len: Optional[int] = None
+) -> str:
+    """
+    Creates a client order id for a new order
+    :param is_buy: True if the order is a buy order, False otherwise
+    :param trading_pair: the trading pair the order will be operating with
+    :param hbot_order_id_prefix: The hummingbot-specific identifier for the given exchange
+    :param max_id_len: The maximum length of the ID string.
+    :return: an identifier for the new order to be used in the client
+    """
+    side = "B" if is_buy else "S"
+    symbols = split_hb_trading_pair(trading_pair)
+    base = symbols[0].upper()
+    quote = symbols[1].upper()
+    base_str = f"{base[0]}{base[-1]}"
+    quote_str = f"{quote[0]}{quote[-1]}"
+    client_instance_id = hex(abs(hash(f"{socket.gethostname()}{os.getpid()}")))[2:6]
```

Comment:
We should keep all the characters from the hash, and truncate it later to take the first N characters being N the number of remaining available spaces based on the max_id_len

Reply:
Addressed as per our discussion